src/mlsquare/architectures/sklearn.py

```diff
@@ -85,6 +87,84 @@ def update_params(self, params):
     def adapter(self):
         return self._adapter
 
+class DimensionalityReductionModel:
+    """
+	A base class for all matrix decomposition models.
+
+    This class can be used as a base class for any dimensionality reduction models.
+    While implementing ensure all required methods are implemented or over written
+    Please refer to sklearn decomposition module for more details.
+
+    Methods
+    -------
+	fit(input_args)
+        fits the model to output singular decomposed values.
+        But outputs an object to further transform.
+
+	fir_transform(input_args)
+        fits the model to output input values with reduced dimensions.
+    """
+    #@abstractmethod
+    #def fit(self, X, y= None, **kwargs):
+    #    """Needs Implementation in sub classes"""
+
+    @abstractmethod
+    def fit_transform(self, X, y=None, **kwargs):
+        """Needs Implementation in sub classes"""
+
+@registry.register
+class SVD(DimensionalityReductionModel, GeneralizedLinearModel):
```

Comment:
then SVD will not need to inherit from GeneralizedLinearModel class

Reply:
But then at line#123, 158,159, it uses GLM's inherited implementation of set_params & update_params?
Should i put similar methods inside above suggested MatrixDecomposition Base class?

Reply:
approach -1: yes. 
a) you can define, BaseTransformer under core. 
b) matrix decompose class, implements the set_params, ge_params like GLM is doing
c) SVD simply inherits from BaseTransformer and MatrixDecompose Classes

Reply:
approach -2:
a) MatrixDecompose inherits BaseModel, implements pass on not-required methods such as a create_model, but defines fit, fit_transform
b) SVD simply inherits from above MatrixDecompose and implements fit_class

Reply:
approach-1 is probably better